TombEngine/Renderer/Renderer11DrawEffect.cpp

```diff
@@ -74,7 +77,163 @@ namespace TEN::Renderer
 		bool IsSoftParticle;
 	};
 
-	void Renderer11::DrawLightning(RenderView& view) 
+	void Renderer11::DrawHelicalLasers(RenderView& view)
+	{
+		byte r, g, b;
+
+		for (int i = 0; i < HelicalLasers.size(); i++)
+		{
+			auto* tg = &HelicalLasers[i];
+
+			if (tg->life)
+			{
+				if (tg->fadein < 8)
+				{
+					r = (tg->r * tg->fadein) >> 3;
+					g = (tg->g * tg->fadein) >> 3;
+					b = (tg->b * tg->fadein) >> 3;
+				}
+				else if (tg->life < 16)
+				{
+					r = (tg->r * tg->life) >> 4;
+					g = (tg->g * tg->life) >> 4;
+					b = (tg->b * tg->life) >> 4;
+				}
+				else
+				{
+					r = tg->r;
+					g = tg->g;
+					b = tg->b;
+				}
+
+				LightningPos[0].x = tg->pos4.x;
+				LightningPos[0].y = tg->pos4.y;
+				LightningPos[0].z = tg->pos4.z;
+
+				memcpy(&LightningPos[1], tg, 48);
```

Comment:
What is this?! It is very unsafe and cryptic operation. Can you make it more clear? I don't understand what is happening here.

Reply:
this function is not necessary for lasers. I removed it